juju/api.go

```diff
@@ -198,6 +200,9 @@ func connectionInfo(args NewAPIConnectionParams) (*api.Info, *jujuclient.Control
 	}
 	account := args.AccountDetails
 	if account.User != "" {
+		if !names.IsValidUserName(account.User) {
+			return nil, nil, errors.NotValidf("user name %q", account.User)
```

Comment:
Please add details of what makes a valid user name.

Reply:
Sure thing.